src/js/actions/layers.js

```diff
@@ -487,12 +488,14 @@ define(function (require, exports) {
                     this.dispatch(events.document.history.nonOptimistic.RESET_BOUNDS, payload);
                 }
             }).then(function () {
+                return this.transfer(guides.queryCurrentGuides);
```

Comment:
Also, do we really need to delay `resetBorderPolicies` until _after_ `queryCurrentGuides` is complete?

Vaguely related gripe: I really want to clean up our action terminology. Quick, what's the difference between "setX", "updateX", "resetX" and "queryX"??